utils/timezone.jsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import UserStore from 'stores/user_store.jsx';
+import * as UserActions from 'actions/user_actions.jsx';
+
+const dateTimeFormat = new Intl.DateTimeFormat();
+
+export function getSupportedTimezones() {
+    if (global.mm_config && global.mm_config.SupportedTimezones) {
+        return global.mm_config.SupportedTimezones.split(',');
+    }
+    return [];
+}
+
+export function getBrowserTimezone() {
+    return dateTimeFormat.resolvedOptions().timeZone;
+}
+
+export function autoUpdateTimezone(userTimezone) {
+    const browserTimezone = getBrowserTimezone();
+    const newTimezoneExists = userTimezone.automaticTimezone !== browserTimezone;
+
+    if (userTimezone.useAutomaticTimezone && newTimezoneExists) {
+        const user = UserStore.getCurrentUser();
+
+        const timezone = {
+            useAutomaticTimezone: 'true',
+            automaticTimezone: browserTimezone,
+            manualTimezone: userTimezone.manualTimezone,
+        };
+
+        const updatedUser = {
+            ...user,
+            timezone,
+        };
+
+        UserActions.updateUser(updatedUser);
+    }
+}
+
+export function getCurrentTimezone(userTimezone) {
```

Comment:
As mentioned above, could this be turned into a selector, possibly combined with the one I was talking about before? Outside of the settings modal, I can't think of any code that will care about the automatic vs manual timezone, so I don't think that needs to be passed around